lib/symboldatabase.cpp

```diff
@@ -5519,7 +5519,16 @@ ValueType ValueType::parseDecl(const Token *type, const Settings *settings)
     return vt;
 }
 
-ValueType::Type ValueType::typeFromString(const std::string &typestr, bool longType)
+ValueType::Type ValueType::wchar_t_type(const Settings *settings)
```

Comment:
Since `settings` is a mandatory argument it could be passed via const reference as well?
To me passing an argument as a pointer always indicates that it could be `NULL`...

Reply:
I'd rather add a check that handles `NULL`, the function works reasonably well anyway.

From a quick grep, it seems every function that takes `Settings` as input argument takes it as a pointer, except the `analyseWholeProgram()` functions.